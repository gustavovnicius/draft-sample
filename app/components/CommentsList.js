import React from 'react'
import _ from 'lodash'
import Comment from 'components/Comment'

const styles = {
  comments: {
    backgroundColor: '#fff',
    padding: '2em',
    boxShadow: '0px 0px 6px 1px rgba(0,0,0,0.18)',
    minHeight: '5vh',
    minWidth: '55vw',
  }
}

const CommentsList = ({comments, onSaveComment, onRemoveComment}) => {
  return (
    <div style={styles.comments}>
      <h3>Comments</h3>
      <div>
        {
          _.map(comments, (comment, key) => {
            return (
              <Comment
                key={key}
                comment={comment}
                onSaveComment={onSaveComment}
                onRemoveComment={onRemoveComment}
              />
            )
          })
        }
      </div>
    </div>
  )
}

export default CommentsList
