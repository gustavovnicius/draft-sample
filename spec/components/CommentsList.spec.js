import React from 'react'
import renderer from 'react-test-renderer'
import CommentsList from 'components/CommentsList.js'

describe('CommentsList', () => {
  it('renders', () => {
    const comments = {
      0: {
        data: {
          comment: '',
          saved: false
        },
        mutability: 'MUTABLE',
        type: 'COMMENT'
      }
    }
    const app = renderer.create(
      <CommentsList comments={comments} />
    )
    const tree = app.toJSON()

    expect(tree).toMatchSnapshot()
  });
});
