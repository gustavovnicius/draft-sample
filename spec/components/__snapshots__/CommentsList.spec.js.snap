exports[`CommentsList renders 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#fff",
      "boxShadow": "0px 0px 6px 1px rgba(0,0,0,0.18)",
      "minHeight": "5vh",
      "minWidth": "55vw",
      "padding": "2em",
    }
  }>
  <h3>
    Comments
  </h3>
  <div>
    <div
      style={
        Object {
          "margin": "2em",
        }
      }>
      <div
        className="Input"
        style={
          Object {
            "boxSizing": "border-box",
            "color": null,
            "marginBottom": 16,
          }
        }>
        <label
          className="Label"
          htmlFor="comment"
          style={
            Object {
              "boxSizing": "border-box",
              "fontSize": 14,
              "fontWeight": 600,
              "lineHeight": 1,
            }
          }>
          
        </label>
        <input
          name="comment"
          onChange={[Function]}
          placeholder="Write a comment"
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, .25)",
              "borderColor": "rgba(0, 0, 0, .25)",
              "borderRadius": 2,
              "borderStyle": "solid",
              "borderWidth": 1,
              "boxSizing": "border-box",
              "color": "inherit",
              "display": "block",
              "fontFamily": "inherit",
              "fontSize": "inherit",
              "height": 32,
              "margin": 0,
              "paddingLeft": 8,
              "paddingRight": 8,
              "width": "100%",
            }
          }
          type="text"
          value="" />
      </div>
      <button
        className="Button"
        href={undefined}
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#08e",
            "border": 0,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "#fff",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "inherit",
            "fontSize": 14,
            "fontWeight": 600,
            "lineHeight": "16px",
            "margin": 0,
            "minHeight": 32,
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
            "textDecoration": "none",
          }
        }>
        Save
      </button>
      <div
        className="Space"
        style={
          Object {
            "boxSizing": "border-box",
            "display": "inline-block",
            "flex": null,
            "width": 8,
          }
        } />
      <button
        className="Button"
        href={undefined}
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#888",
            "border": 0,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "#fff",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "inherit",
            "fontSize": 14,
            "fontWeight": 600,
            "lineHeight": "16px",
            "margin": 0,
            "minHeight": 32,
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
            "textDecoration": "none",
          }
        }>
        Remove
      </button>
    </div>
  </div>
</div>
`;
